zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/RoleTest.java

```diff
@@ -273,4 +281,47 @@ public void shouldRejectIfEntityIsNotPresentEntityRemoval() {
             "Expected to remove an entity with key '%s' and type '%s' from role with key '%s', but the entity doesn't exist."
                 .formatted(1L, EntityType.USER, roleKey));
   }
+
+  @Test
+  public void shouldDeleteRole() {
+    // given
+    final var name = UUID.randomUUID().toString();
+    final var roleKey = ENGINE.role().newRole(name).create().getValue().getRoleKey();
+    ENGINE
+        .authorization()
+        .permission()
+        .withOwnerKey(roleKey)
+        .withOwnerType(AuthorizationOwnerType.ROLE)
+        .withResourceType(AuthorizationResourceType.ROLE)
+        .withAction(PermissionAction.REMOVE)
+        .add();
+
+    // when
+    ENGINE.role().deleteRole(roleKey).withName(name).delete();
+
+    final var deletedRecord = roleState.getRole(roleKey);
+    assertTrue(deletedRecord.isEmpty());
+  }
+
+  @Test
+  public void shouldRejectIfRoleIsNotPresentOnDeletion() {
+    // given
+    final var name = UUID.randomUUID().toString();
+    final var roleRecord = ENGINE.role().newRole(name).create();
```

Comment:
❌ I don't see a reason to create a role here. The test wants to verify the command is rejected if the role doesn't exist. Why do we need this role?

Reply:
It's just testing purpose, but yes it can be removed 👍